### PR TITLE
Upgrade laminas-coding-standard to 2.2

### PR DESCRIPTION
**Pull request by arueckauer**
|    Q          |   A
|-------------- | ------
| QA            | yes

### Description

This PR updates the `laminas-coding-standard` to version `2.2.1` and provides necessary code changes.

For the upgrade to `2.3.0`, "only" strict type declaration is required. I am not sure, if that would break parts of the component.

### CODE REVIEW

**Comment by arueckauer** [comment]
QA check phpcs fails due to locked `laminas/laminas-code` version `4.2.0` [no longer supporting PHP 7.3](https://github.com/laminas/laminas-code/blob/4.2.0/composer.json). Change was introduced in [c2b9fba](https://github.com/laminas/laminas-xmlrpc/commit/9181b9f967f6c3ce19900d7bbfbe47b25b7308d2#diff-f37acfaa6b11f575a9a6f41a75fa73a61d0f8ebc2c9b8cddc215d8aca10e44f5).

Would raising mimimum PHP version to 7.4 be an appropriate solution?

**Comment by froschdesign** [comment]
@arueckauer 
> Would raising mimimum PHP version to 7.4 be an appropriate solution?

PHP 7.3 can be dropped with a new minor version and [the end of 7.3 is near](https://www.php.net/supported-versions).

**Comment by froschdesign** [comment]
@arueckauer 

```
Error: docs/book/server.md:136:1 MD004/ul-style Unordered list style [Expected: asterisk; Actual: dash]
Error: docs/book/server.md:137:1 MD004/ul-style Unordered list style [Expected: asterisk; Actual: dash]
Error: docs/book/server.md:138:1 MD004/ul-style Unordered list style [Expected: asterisk; Actual: dash]
```

~Use dashes:~ Use asterisks: 

```markdown
* `system.listMethods`
* `system.methodHelp`
* `system.methodSignature`
```

https://github.com/laminas/laminas-xmlrpc/blob/2.11.x/docs/book/server.md

File: src/Response/Http.php [hunk 1]
```diff
@@ -23,7 +20,7 @@ class Http extends XmlRpcResponse
     public function __toString()
     {
         if (! headers_sent()) {
-            header('Content-Type: text/xml; charset=' . strtolower($this->getEncoding()));
```
Comment (Ocramius) [inline src/Response/Http.php]: This change seems incorrect/risky

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: src/Response/Http.php (hunk 1 above)
Comment (froschdesign) [reply src/Response/Http.php]: The default value is:

https://github.com/laminas/laminas-xmlrpc/blob/b29dc77fa720147f6fff83d7ebbb3375f4e8ace4/src/Response.php#L32-L36

So this change produces a different result.

Comment (Ocramius) [reply src/Response/Http.php]: Yes, but then this used to produce `charset=utf-8`, which is what was expected when casting this header to `(string)`.

We shouldn't change this.

Comment (froschdesign) [reply src/Response/Http.php]: That's what I mean! 😃

Comment (arueckauer) [reply src/Response/Http.php]: You are absolutely right, this was not meant to be in here.

**Review by Ocramius — APPROVED** [review]
Thanks @arueckauer!

**Pull request merged**